views/hooks/useChartsList.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import { useOutletContext } from 'react-router-dom'
+import api from '../api'
+
+const NULL_CHART = {}
+
+export default function useChartsList() {
+  const { user, setNotification, users } = useOutletContext()
+  const [chartToShare, setChartToShare] = React.useState(NULL_CHART)
+  const [chartList, setChartList] = React.useState([])
+  const [usernames, setUsernames] = React.useState([])
+
+  const closeDialog = () => setChartToShare(NULL_CHART)
+  const onShare = (chart) => setChartToShare(chart)
+  const shareWithUsers = async (chart_id, sharedWith, options = {}) => {
+    try {
+      await api.charts.chartsShare.create(chart_id, sharedWith)
+
+      const updatedChart = { ...chartToShare, sharedWith }
+      const updatedChartList = chartList.map((chart) =>
+        chart._id === updatedChart._id ? updatedChart : chart
+      )
+
+      setChartList(updatedChartList)
+      options.closeDialog ? closeDialog() : setChartToShare(updatedChart)
+      setNotification({ open: true, message: 'Shared chart with user' })
+    } catch (error) {
+      setNotification({ open: true, message: error.message })
+    }
+  }
+  const onDelete = async (chart) => {
+    try {
+      await api.charts.chart.destroy(chart._id)
+
+      await loadCharts()
+    } catch (error) {
+      setNotification({ open: true, message: error.message })
+    }
+  }
+  const onDuplicate = async (chart) => {
+    try {
+      await api.charts.chartsDuplicate.create(chart._id)
+
+      await loadCharts()
+    } catch (error) {
+      setNotification({ open: true, message: error.message })
+    }
+  }
+  const loadCharts = async () => {
+    try {
+      const data = await api.charts.chart.index()
+
+      setChartList(data)
+    } catch (error) {
+      setNotification({ open: true, message: error.message })
+    }
+  }
+
+  React.useEffect(() => {
```

Comment:
Same as above but with `useEffect`, we don't need to dig into the React object.

Reply:
[f69e1dd](https://github.com/AMP-SCZ/dpdash/pull/500/commits/f69e1dd2e2176c10c6ff393ebef0663883ac1cdd)